src/tests/Directory.Build.targets

```diff
@@ -34,8 +34,8 @@
 
   <!-- Default priority building values. -->
   <PropertyGroup>
-    <DisableProjectBuild Condition="'$(IsMergedTestRunnerAssembly)' == 'true' and $(BuildAsStandalone)">true</DisableProjectBuild>
-    <OutputType Condition="('$(IsMergedTestRunnerAssembly)' == 'true' and !$(BuildAsStandalone)) or ('$(RequiresProcessIsolation)' == 'true' and '$(CLRTestKind)' != 'SharedLibrary')">Exe</OutputType>
+    <DisableProjectBuild Condition="'$(IsMergedTestRunnerAssembly)' == 'true' and '$(BuildAllTestsAsStandalone)' and '$(HasMergedInTests)' != 'true'">true</DisableProjectBuild>
```

Comment:
Seems like the semantic problem we're observing is because MSBuild is expecting `BuildAsStandalone` to be explicitly compared to `'true'` or '`false'`.